doc/source/data/dataset.rst

```diff
@@ -377,7 +377,149 @@ Tensor datasets can also be created from NumPy ndarrays that are already stored
     arr2 = np.arange(10, 20)
     ds = ray.data.from_numpy([ray.put(arr1), ray.put(arr2)])
 
-Limitations: currently tensor-typed values cannot be nested in tabular records (e.g., as in TFRecord / Petastorm format). This is planned for development.
+Tables with tensor columns
+--------------------------
+
+In addition to tensor datasets, Datasets also supports tables with fixed-shape tensor columns, where each element in the column is a tensor (n-dimensional array) with the same shape. As an example, this allows you to use both Pandas and Ray Datasets to read, write, and manipulate a table with a column of e.g. images (2D arrays), with all conversions between Pandas, Arrow, and Parquet, and all application of aggregations/operations to the underlying image ndarrays, being taken care of by Ray Datasets.
+
+With our Pandas extension type, :class:`TensorDtype <ray.data.extensions.tensor_extension.TensorDtype>`, and extension array, :class:`TensorArray <ray.data.extensions.tensor_extension.TensorArray>`, you can do familiar aggregations and arithmetic, comparison, and logical operations on a DataFrame containing a tensor column and the operations will be applied to the underlying tensors as expected. With our Arrow extension type, :class:`ArrowTensorType <ray.data.extensions.tensor_extension.ArrowTensorType>`, and extension array, :class:`ArrowTensorArray <ray.data.extensions.tensor_extension.ArrowTensorArray>`, you'll be able to import that DataFrame into Ray Datasets and read/write the data from/to the Parquet format. Automatic conversion between the Pandas and Arrow extension types/arrays keeps the details under-the-hood, so you only have to worry about casting the column to a tensor column using our Pandas extension type when creating the Pandas DataFrame or when reading table from storage (if the table wasn't saved by Ray Datasets and instead contains bytes blobs for each ndarray in the tensor column); everything downstream from that casting should work automatically.
+
+.. code-block:: python
+
+    import numpy as np
```

Comment:
How about organizing this into a few sections in this order:
- Read an existing 'bytes' column in parquet as a Tensor type
- Working with tensor type columns
- Write tensor types using pandas metadata
- Converting to and from pandas tensor type

Reply:
Btw, the reason to order it as the above is
(1) breaking the code block up into smaller sections will be more digestable to users
(2) users onboarding will want to address concerns in the above order, not the order currently presented (which starts with creating tensor datasets in pandas)